src/rust/lib/graph-editor/src/component/operator.rs

```diff
@@ -0,0 +1,271 @@
+        self.disable_fullscreen();
+        let operator = FullscreenOperator::apply(target,scene);
+        self.operator.set(operator);
+    }
+
+    /// Disables fullscreen mode for the given component.
+    pub fn disable_fullscreen(&self) {
+        if let Some(old) = self.operator.borrow_mut().take() {
+            old.undo();
+        }
+    }
+
+    /// Return a ref clone of the fullscreen element.
+    pub fn get_element(&self) -> Option<T>{
+        self.operator.borrow().as_ref().map(|op| op.target.clone_ref())
+    }
+}
+
+
+
+// ===============================
+// === Layer Management Helper ===
+// ===============================
+
+/// Indicates the required target layer.
+// FIXME this is a layer management hack. Remove this once we have nicer scene layer management.
+#[derive(Debug)]
+pub enum SymbolType {
+    /// A symbol that goes onto the `Main` layer.
+    Main (Symbol),
+    /// A visualisation symbol that goes above the `Main` layer, but below the cursor.
+    Visualisation (Symbol),
+}
+
+/// Moves the given components shapes to the default scene layers.
+// FIXME This is an ugly hack for layer management.
+// FIXME Needs to be removed as soon as we have something better.
+pub fn set_layers_normal<T: NativeComponent>(target:&T, scene:&Scene){
+    target.unset_layers_all(&scene);
+    for symbol in target.symbols() {
+        match symbol {
+            SymbolType::Main(symbol)          => scene.views.main.add(&symbol),
+            SymbolType::Visualisation(symbol) => scene.views.visualisation.add(&symbol),
+        }
+    }
+}
+
+/// Moves the given components shapes to the fullscreen scene layers.
+// FIXME This is an ugly hack for layer management.
+// FIXME Needs to be removed as soon as we have something better.
+pub fn set_layers_fullscreen<T: NativeComponent>(target:&T, scene:&Scene) {
+    target.unset_layers_all(&scene);
+    for symbol in target.symbols() {
+        match symbol {
+            SymbolType::Main(symbol)          => scene.views.overlay.add(&symbol) ,
+            SymbolType::Visualisation(symbol) => scene.views.overlay_visualisation.add(&symbol) ,
+        }
+    }
+}
+
+
+
+// ===========================
+// === Fullscreen Operator ===
+// ===========================
+
+pub trait Fullscreenable = display::Object+Resizable+NativeComponent+Networked+CloneRef+'static;
+
+/// A `FullscreenOperator` can be used to apple fullscreen mode to a UI element as well as undo the
+/// undo the fullscreen operation and restore the previous state. The  `FullscreenOperator` can be
+/// applied to any target that implements `display::Object`, `Resizable` and `NativeUiElement`.
+// TODO consider incorporating these traits into display::Object or another common "SceneElement"
+// type. But it is important that complex UI components can provide information about their
+// sub-components (for example, multiple sub-shapes or HTML components).
+#[derive(Debug)]
+pub struct FullscreenOperator<T> {
```

Comment:
This whole thing works currently on visualizations only, lets move it to `component/visualization/fullscreen` for now, until we see a generalization pattern.